frontend/packages/console-demo-plugin/src/plugin.ts

```diff
@@ -1,23 +1,33 @@
 import {
   Plugin,
+  ModelFeatureFlag,
   HrefNavItem,
   ResourceNSNavItem,
+  ResourceClusterNavItem,
   ResourceListPage,
   ResourceDetailPage,
-  ModelFeatureFlag,
 } from '@console/plugin-sdk';
 
 // TODO(vojtech): internal code needed by plugins should be moved to console-shared package
 import { PodModel } from '@console/internal/models';
+import { FLAGS } from '@console/internal/const';
 
 type ConsumedExtensions =
+  | ModelFeatureFlag
   | HrefNavItem
   | ResourceNSNavItem
+  | ResourceClusterNavItem
   | ResourceListPage
-  | ResourceDetailPage
-  | ModelFeatureFlag;
+  | ResourceDetailPage;
 
 const plugin: Plugin<ConsumedExtensions> = [
+  {
+    type: 'FeatureFlag/Model',
```

Comment:
The reason for moving this extension up in the list is to highlight that each plugin should start with adding its own feature flag(s).

Other extensions like nav items should be gated by those feature flags.